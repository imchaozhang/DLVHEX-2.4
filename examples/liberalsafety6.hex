% If evaluated with --heuristics=monolithic and --domainexpansionsafety,
% this program tests if it works that outer external atoms become inner ones.
% While forbidden with strong safety, this should work with domainexpansionsafety.

% If evaluated with --heuristics=monolithic, parameter q in &id[q](Z) is unstratified,
% thus the domain exploration algorithm needs to guess the assignments to q(a).
% With --heuristics=greedy the second rule is in a different component and
% both can be solved with plain model generator.
p(a).
q(Y) :- p(X), &testConcat[X,a](Y).
res(Z) :- &id[q](Z).
