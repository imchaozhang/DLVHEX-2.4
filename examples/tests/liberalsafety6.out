{p(a),q(aa),res(aa)}
